src/curl.c

```diff
@@ -93,16 +101,24 @@ int swupd_curl_init(void)
 				continue;
 			}
 			fallback_capaths[i] = strdup(tok);
-			if (++i > CA_PATH_NO_MAX) {
-				fprintf(stderr, "Warning: up to %d CApaths can be specified."
-						" Ignoring the rest.\n", CA_PATH_NO_MAX);
-				break;
+			if (++i > sz) {
+				char **re_fallback_capaths, **t;
+				sz <<= 1;
+				re_fallback_capaths = (char **)malloc(sz*sizeof(char *));
```

Comment:
Why do we care about malloc()'s return value above this, but not the two spots below?